src/sentry/services/hybrid_cloud/project_key/impl.py

```diff
@@ -27,6 +28,20 @@ def get_project_key(
     ) -> Optional[RpcProjectKey]:
         return self._get_project_key(project_id=project_id, role=role)
 
+    def get_default_project_key(
+        self, *, organization_id: int, project_id: str
+    ) -> Optional[RpcProjectKey]:
+        from sentry.models.project import Project
+        from sentry.models.projectkey import ProjectKey
+
+        try:
+            project = Project.objects.get(id=project_id)
```

Comment:
This unnecessary `Project` query isn't great, but it's necessary to delegate to `ProjectKey.get_default` as-is. The alternative would be to duplicate some code from

```python
    @classmethod
    def get_default(cls, project):
        return cls.objects.filter(
            project=project,
            roles=models.F("roles").bitor(cls.roles.store),
            status=ProjectKeyStatus.ACTIVE,
        ).first()
```

into

```python
        key = ProjectKey.objects.filter(
            project_id=project_id,
            roles=models.F("roles").bitor(ProjectKey.roles.store),
            status=ProjectKeyStatus.ACTIVE,
        ).first()
        return serialize_project_key(key) if key else None
```

Reply:
You could use `Project.objects.get_from_cache(id=project_id)` to skip a DB query *most* of the time.